### PR TITLE
ignoring deprecation warning on NSUnarchiver

### PR DESCRIPTION
**Pull request by Coeur**
Problem: NSUnarchiver is deprecated but we need it for data archived with Transmission 3
Solution: We silence the warning for as long as Apple supports this class

### CODE REVIEW

File: macosx/NSKeyedUnarchiverAdditions.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// This file Copyright © 2022-2022 Transmission authors and contributors.
```
Comment (nevack) [inline macosx/NSKeyedUnarchiverAdditions.h:1]: I don't think we need year-range here, just `2022` will do.

File: macosx/SparkleProxy.mm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// This file Copyright © 2005-2022 Transmission authors and contributors.
+// This file Copyright © 2022-2022 Transmission authors and contributors.
```
Comment (nevack) [inline macosx/SparkleProxy.mm:1]: Again here, I don't think we should update copyright year

File: macosx/CocoaCompatibility.h [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// This file Copyright © 2011-2022 Transmission authors and contributors.
+// This file Copyright © 2022-2022 Transmission authors and contributors.
```
[the remark this answers is not in the document]
Comment (Coeur) [reply macosx/CocoaCompatibility.h:1]: This file and content was created by me this year.

File: macosx/SparkleProxy.mm (hunk 2 above)
Comment (Coeur) [reply macosx/SparkleProxy.mm:1]: This file and content was created by me this year.

**Review by nevack — APPROVED** [review]
LGTM, thanks @Coeur !

**Pull request merged**